src/coreclr/inc/corinfo.h

```diff
@@ -1649,6 +1649,7 @@ enum CORINFO_FIELD_ACCESSOR
     CORINFO_FIELD_STATIC_ADDR_HELPER,       // static field accessed using address-of helper (argument is FieldDesc *)
     CORINFO_FIELD_STATIC_TLS,               // unmanaged TLS access
     CORINFO_FIELD_STATIC_READYTORUN_HELPER, // static field access using a runtime lookup helper
+    CORINFO_FIELD_STATIC_DATASEGMENT,       // static field access from the data segment
```

Comment:
We have documentation [above](https://github.com/dotnet/runtime/blob/3bf9945955465fb5cbc692779358d69621b37c03/src/coreclr/inc/corinfo.h#L116) on what the various accessors mean, it would be nice to update it to cover this new value.

(I realize some of it is rather outdated)